File: miscellaneous/sorting/bubble-sort/20230224/BubbleSortTest.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;
require __DIR__ .'/bubble-sort.php';

class BubbleSortTest extends TestCase
{
    private array $testCases;

    public function __construct()
    {
        parent::__construct();

        $this->testCases = [
            [],
            [2],
            [2, 1],
            [99, 44, 6, 2],
            [1, 2, 3, 4, 5, 6, 9, 100],
            [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0],
        ];
    }

    public function test_bubble_sort()
    {
        array_map(function ($nums) {
            $sorted = $nums;
            sort($sorted);
            $this->assertEquals(expected: $sorted, actual: bubbleSort($nums));
        }, $this->testCases);
    }
}

File: miscellaneous/sorting/bubble-sort/20230224/bubble_sort_test.py
from bubble_sort import bubble_sort

test_cases = [
    [],
    [2],
    [2, 1],
    [99, 44, 6, 2],
    [1, 2, 3, 4, 5, 6, 9, 100],
    [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0],
]


def test_bubble_sort():
    for nums in test_cases:
        assert sorted(nums) == bubble_sort(nums)